pipeline-yaml_5b5744d40428631d7a8940a9.md

```diff
@@ -0,0 +1,548 @@
+            - echo $SEMAPHORE_PIPELINE_ID
+
+## The secrets property
+
+The `secrets` property is for using existing `secrets` environment variables.
+
+Please notice that when the names of the environment variables of two
+more more `secrets` are the same, then the environment variable will
+have the value that can be found in the `secrets` property that was
+imported last.
+
+Additionally, if you try to use a `name` value that does not exist, the
+pipeline will fail to compile and therefore run.
+
+
+### name
+
+The `name` property is compulsory is a `secrets` block because it
+specifies the `secrets` bucket that you want to use. Please note
+that the specified `secrets` bucket should be under the
+active organization.
+
+
+### env_var_names
+
+The `env_var_names` property lets you select the environment variables
+that you want to use from the specified `secrets` bucket. Currently only
+the `[]` value is accepted, which loads all environment variables found under
+the specified  `secrets` bucket.
+
+
+### Example of secrets
+
+      secrets:
+        - name: mysql-secrets
+          env_var_names: []
+
+Note that the environment variable defined in a `secrets` property are used
+like regular environment variables defined in a `env_vars` block.
+
+
+## Complete Configuration examples
+
+
+###  A complete .semaphore/semaphore.yml example
+
+The following code presents a complete `.semaphore/semaphore.yml` file:
+
+    version: "v1.0"
+    name: YAML file example for Go project.
+    agent:
+      machine:
+        type: e1-standard-2
+        os_image: ubuntu1804
+    blocks:
+     - name: Inspect Linux environment
+       task:
+          jobs:
+            - name: Execute hw.go
+              commands:
+                - echo $SEMAPHORE_PIPELINE_ID
+                - echo $HOME
+                - echo $SEMAPHORE_GIT_DIR
+                - pwd
+                - ls -al
+                - cat hw.go
+                - echo $VAR1
+                - echo $PI
+          prologue:
+              commands:
+                - checkout
+          env_vars:
+               - name: VAR1
+                 value: Environment Variable 1
+               - name: PI
+                 value: "3.14159"
+
+     - name: Calling text file
+       task:
+          jobs:
+            - name: Using command_file
+              cmd_file: command_file
+          prologue:
+              commands:
+                - checkout
+
+     - name: Build Go project
+       task:
+          jobs:
+            - name: Build hw.go
+              commands:
+                - checkout
+                - sudo apt-get -y install golang-go
+                - go build hw.go
+                - ls -l ./hw
+                - file ./hw
+                - ./hw
+            - name: PATH variable
+              commands:
+                - echo $PATH
+          epilogue:
+              commands:
+                - echo $SEMAPHORE_JOB_RESULT
+                - echo $SEMAPHORE_PIPELINE_ID
```

Comment:
Make this example less verbose and shorter.

Reply:
Done!